### PR TITLE
bfix: handles error incase of unsuccessful logout

### LINKED ISSUE #2416: logout button lands to Select Language page even when logout is not successful.
**Describe the bug**
When one clicks on Logout button, it will redirect user to the Select Language screen, even when the logout was not successful due to any error occurring back of the scene.

**To Reproduce**

1. Login into an account and go to the settings option on the profile page
2. Turn off the internet connection.
3. Logout the account. This will navigate to the Select Language Screen.
4. Select a language and you will get navigated to the settings screen again, despite being logged out.

**Expected behavior**
The logout should throw an error as the logout is not successful.

**Actual behavior**
The logout is shown as successful and we are navigated to the Select Language screen

### PR DESCRIPTION
**Pull request by am-casper**
**What kind of change does this PR introduce?**
This fixes a bug. Error handing in case of unsuccessful logging out is improved.

**Issue Number:** #2416 

Fixes #2416 

**Did you add tests for your changes?**

Updated a couple of tests.

**Summary**

As a user, i don't want to not able to log out properly and stuck between the App Settings and Select Language screen, everytime a logout is unsuccessful. 

**Does this PR introduce a breaking change?**

This fixes a critical bug, especially for users who have unstable internet connection.

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

**Comment by dhanagopu** [comment]
@CyberWake @Kevoniat  Can you review this PR?

**Comment by noman2002** [comment]
@am-casper Please attach a video.

**Comment by am-casper** [comment]
This is after the bug fix. Before the bug was fixed, app crashes after logout if internet connectivity is weak
https://github.com/PalisadoesFoundation/talawa/assets/103204377/ba41e69b-c28b-4e2a-a79a-5f6258877c61

**Review by noman2002 — CHANGES_REQUESTED** [review]
When logged out, navigate it to the login screen not on the language selection screen.

**Comment by am-casper** [comment]
> When logged out, navigate it to the login screen not on the language selection screen.

@noman2002 updated the commit. Kindly have a look

File: lib/views/after_auth_screens/app_settings/app_settings_page.dart [hunk 1]
```diff
@@ -302,11 +302,16 @@ class AppSettingsPage extends StatelessWidget {
                         successText: 'LogOut',
                         success: () async {
                           try {
-                            await model.logout();
+                            final bool isLogoutSuccessful =
+                                await model.logout();
+                            if (!isLogoutSuccessful) {
+                              throw Error(); //checks whether the logout was successful or not.
```
Comment (literalEval) [inline lib/views/after_auth_screens/app_settings/app_settings_page.dart:308]: Where is this error catched ?

File: lib/view_model/after_auth_view_models/settings_view_models/app_setting_view_model.dart [hunk 2]
```diff
@@ -14,10 +14,11 @@ class AppSettingViewModel extends BaseModel {
   ///   None
   ///
   /// **returns**:
-  ///   None
-  Future<void> logout() async {
+  /// * `Future<bool>`: A [Future] that resolves to a [bool] value indicating whether the user has been logged out.
```
Comment (literalEval) [inline lib/view_model/after_auth_view_models/settings_view_models/app_setting_view_model.dart:17]: Write a small non-redundant sentence.
Like: "Logs the user out and returns logout status"

**Review by literalEval — CHANGES_REQUESTED** [review]

File: lib/views/after_auth_screens/app_settings/app_settings_page.dart (hunk 1 above)
Comment (am-casper) [reply lib/views/after_auth_screens/app_settings/app_settings_page.dart:308]: @literalEval there's a catch block at line 316. please have a look at it.

File: lib/view_model/after_auth_view_models/settings_view_models/app_setting_view_model.dart (hunk 2 above)
Comment (am-casper) [reply lib/view_model/after_auth_view_models/settings_view_models/app_setting_view_model.dart]: @literalEval updated.

**Comment by palisadoes** [comment]
This is an update on the PR merging freeze:

1. In the next few hours we will be merging the `develop-userTypeFix` branch into the `develop` branch. 
1. The `develop-userTypeFix` branch was created to fix a long standing design flaw where Admins were Admins of all organizations, not specific ones.
1. The `userType` field has been removed from the User collection and it has been replaced by an `appUserProfileId` field.
    1. This field is `null` if the user isn’t registered to use the apps. This will help people to add users manually during the event checkin process, or if an Admin wants to manually add someone in the Admin dashboard.
    1. When not `null` the `AppUserProfileID` will reference a `AppUserProfile` collection with App related information such as the organizations for which a user may be an Admin.
    1. The updated schema can be found here https://github.com/PalisadoesFoundation/talawa-api/blob/develop-userTypeFix/schema.graphql
    1. This is the parent issue that we have been using to track progress:
        1. https://github.com/PalisadoesFoundation/talawa-api/issues/1965
1. This merge will cause some conflicts in your PR.

We decided to do this at the beginning of the weekend to give us all time to adjust PR code and create bug fixes that may arise.

Update your code at or after midnight GMT on the morning of March 23, 2024. (5:30am IST). 

If your PRs have already been approved, request a re-review after fixing the conflicts and refactoring to the new `AppUserProfileID` methodology.

**Review by noman2002 — APPROVED** [review]
Looks fine to me.

**Pull request merged**